cmd/frontend/internal/app/editor.go

```diff
@@ -82,38 +83,89 @@ type editorOpenFileRequest struct {
 // editorSearchRequest represents parameters for "search on Sourcegraph" editor requests.
 type editorSearchRequest struct {
 	query string // The literal search query
+
+	// Optional git repository remote URL. When present, the search will be performed just
+	// in the repository (not globally).
+	remoteURL         string
+	hostnameToPattern map[string]string // Map of Git remote URL hostnames to patterns describing how they map to Sourcegraph repositories
+
+	// Optional git repository branch name and revision. When one is present and remoteURL
+	// is present, the search will be performed just at this branch/revision.
+	branch, revision string
```

Comment:
We don't have lint enabled right now, but I think it will complain.
```suggestion
	branch   string
	revision string
```